pkg/controller/statefulset/stateful_set.go

```diff
@@ -419,7 +419,7 @@ func (ssc *StatefulSetController) sync(key string) error {
 func (ssc *StatefulSetController) syncStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
 	glog.V(2).Infof("Syncing StatefulSet %v/%v with %d pods", set.Namespace, set.Name, len(pods))
 	if err := ssc.control.UpdateStatefulSet(set, pods); err != nil {
-		glog.V(2).Infof("Error syncing StatefulSet %s/%s with %d pods : %s", set.Namespace, set.Name, err)
+		glog.V(2).Infof("Error syncing StatefulSet %s/%s with %d pods : %s", set.Namespace, set.Name, len(pods), err)
```

Comment:
You could handle the error statement in the outer loop, having the statement does allow for symmetry in the log statements. You'll see the an entering, and you get a log statement no matter which control path the function takes. Bumping the log level to 4 makes sense at the error is already reported if it occurs.